packages/cli/src/lib/emitUsage.ts

```diff
@@ -0,0 +1,77 @@
+import { Usage } from '@appland/client';
+import { Metadata } from '@appland/models';
+import { Git } from '../telemetry';
+import type { UsageReport, UsageUpdateDto } from '@appland/client/dist/src/usage';
+import sanitizeURL from './repositoryInfo';
+
+function serializeMetadata(metadata?: Metadata): string | undefined {
+  if (!metadata) return;
+
+  return JSON.stringify({
+    app: metadata.app,
+    language: metadata.language,
+    client: metadata.client,
+    frameworks: metadata.frameworks,
+    git: {
+      repository: metadata.git?.repository ? sanitizeURL(metadata.git.repository) : undefined,
+      branch: metadata.git?.branch,
+      commit: metadata.git?.commit,
+    },
+    recorder: metadata.recorder,
+  });
+}
+
+let usageReport: UsageReport | undefined;
+async function getCycleStartDate(): Promise<Date> {
+  const now = new Date();
+
+  if (usageReport?.cycleEnd && now > usageReport.cycleEnd) {
+    // We've surpassed the end of the current cycle. By invalidating the
+    // cached usage report, we'll fetch a new one from the server.
+    usageReport = undefined;
+  }
+
+  if (!usageReport) {
+    // We don't have a cached usage report, so fetch one from the server
+    try {
+      usageReport = await Usage.get();
+    } catch (e) {
+      // Do nothing, we'll try again later.
+      // At the time being, we don't need to rate limit the client.
+      // The client will either only emit usage once per execution, or
+      // at a maximum, once per 10 seconds.
```

Comment:
Log something? Completely swallowed exceptions freak me out.

Reply:
I added a message when running in verbose mode.